lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -420,55 +434,72 @@ export class UserAgentApplication {
         scopes = [this.clientId];
       }
 
-      const scope = scopes.join(" ").toLowerCase();
-      var popUpWindow = this.openWindow("about:blank", "_blank", 1, this, resolve, reject);
-      if (!popUpWindow) {
-        return;
-      }
-    
-      this._loginInProgress = true;
-
-      this.authorityInstance.ResolveEndpointsAsync().then(() => {
-        const authenticationRequest = new AuthenticationRequestParameters(this.authorityInstance, this.clientId, scopes, ResponseTypes.id_token, this._redirectUri, this._state);
-        if (extraQueryParameters) {
-          authenticationRequest.extraQueryParameters = extraQueryParameters;
+        const adalIdToken = this._cacheStorage.getItem(Constants.adalIdToken);
+        var idTokenObject;
+        if (adalIdToken) {
+            idTokenObject = Utils.extractIdToken(adalIdToken);
         }
-
-        this._cacheStorage.setItem(Constants.loginRequest, window.location.href, this.storeAuthStateInCookie);
-        this._cacheStorage.setItem(Constants.loginError, "");
-        this._cacheStorage.setItem(Constants.nonceIdToken, authenticationRequest.nonce, this.storeAuthStateInCookie);
-        this._cacheStorage.setItem(Constants.msalError, "");
-        this._cacheStorage.setItem(Constants.msalErrorDescription, "");
-        const authorityKey = Constants.authority + Constants.resourceDelimeter + authenticationRequest.state;
-        if (Utils.isEmpty(this._cacheStorage.getItem(authorityKey))) {
-          this._cacheStorage.setItem(authorityKey, this.authority);
+        if (idTokenObject && idTokenObject.upn) {
+            console.log("ADAL's idToken exists. Extracting login information from ADAL's idToken ");
+            var extraQueryParam= "&" + Constants.login_hint+ "=" + idTokenObject.upn + "&" + Constants.domain_hint + "=" + Constants.organizations;
```

Comment:
Do you require extraQueryparams to start with "&"?

Reply:
Yes. We have a separate login else where to make sure it starts with "?"

  // if the endpoint already has queryparams, lets add to it, otherwise add the first one
        if (authEndpoint.indexOf("?") < 0) {
            authEndpoint += "?";
        } else {
            authEndpoint += "&";
        }